synapse/replication/http/_base.py

```diff
@@ -193,23 +206,26 @@ async def send_request(instance_name="master", **kwargs):
                     try:
                         result = await request_func(uri, data, headers=headers)
                         break
-                    except CodeMessageException as e:
-                        if e.code != 504 or not cls.RETRY_ON_TIMEOUT:
+                    except RequestTimedOutError:
+                        if not cls.RETRY_ON_TIMEOUT:
                             raise
 
-                    logger.warning("%s request timed out", cls.NAME)
+                    logger.warning("%s request timed out; retrying", cls.NAME)
 
                     # If we timed out we probably don't need to worry about backing
                     # off too much, but lets just wait a little anyway.
                     await clock.sleep(1)
             except HttpResponseException as e:
                 # We convert to SynapseError as we know that it was a SynapseError
-                # on the master process that we should send to the client. (And
+                # on the main process that we should send to the client. (And
                 # importantly, not stack traces everywhere)
+                _outgoing_request_counter.labels(cls.NAME, e.code).inc()
                 raise e.to_synapse_error()
-            except RequestSendFailed as e:
-                raise SynapseError(502, "Failed to talk to master") from e
+            except Exception as e:
+                _outgoing_request_counter.labels(cls.NAME, "ERR").inc()
+                raise SynapseError(502, "Failed to talk to main process") from e
 
+            _outgoing_request_counter.labels(cls.NAME, 200).inc()
```

Comment:
We've never used `200` as a label before, but I suppose this is just to differentiate between the other groups. I assume we can't just leave out the `200` here as otherwise we wouldn't be able to filter by "successful" requests?

Reply:
uhhh, I don't think I understand your question.

The idea is that  `synapse_outgoing_replication_requests` tracks the number of requests, broken down by endpoint (`cls.NAME`) and response code. For non-200 response codes, the counter is bumped at line 222. For successful responses, there's a different code path, so we bump the counter here.

We've got to put *something* in that label, so we can't just leave out the `200`.